api/jsonrpc/client.go

```diff
@@ -194,21 +194,25 @@ func (cli *JSONRPCClient) GetABI(ctx context.Context) (abi.ABI, error) {
 	return resp.ABI, err
 }
 
-func (cli *JSONRPCClient) Execute(ctx context.Context, actor codec.Address, action chain.Action) ([]byte, error) {
-	actionBytes, err := chain.MarshalTyped(action)
-	if err != nil {
-		return nil, fmt.Errorf("failed to marshal action: %w", err)
+func (cli *JSONRPCClient) Execute(ctx context.Context, actor codec.Address, actions []chain.Action) ([][]byte, error) {
+	actionsMarshaled := make([][]byte, 0)
```

Comment:
nit: can we pre-allocate where we already know the length of the slice? Same comment throughout